### PR TITLE
[stable/redis] Allow slave pods to connect to master when networkPolicy is enabled

### LINKED ISSUE #14915: [stable/redis] networkPolicy blocks slave pods from connecting
**Describe the bug**
Enabling networkPolicy blocks the slave pods from connecting to master as the slave pods are missing the pod selector label (e.g. redis-client=true) 

**Version of Helm and Kubernetes**:
Helm: v2.14.1
Kubernetes: v1.12.7

**Which chart**:
Redis

**What happened**:
Redis-slave pods are unhealthy and restarts constantly as it cannot connect to master

**What you expected to happen**:
Redis-slave pods come up clean

**How to reproduce it** :
```helm install stable/redis --values values-production.yaml``` (or networkPolicy.enabled=true)

### PR DESCRIPTION
**Pull request by Yitaek**
#### What this PR does / why we need it:
@miguelaeh Here's the PR to allow slave pods to connect to master pod 

#### Which issue this PR fixes
  - fixes #14915 

#### Checklist
- [x ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x ] Chart Version bumped
- [x ] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/redis/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -31,6 +31,13 @@ spec:
             release: "{{ .Release.Name }}"
             role: metrics
             app: {{ template "redis.name" . }}
+      {{- if ge .Values.cluster.slaveCount 1.0 }}
+      - podSelector:
+          matchLabels:
+            release: "{{ .Release.Name }}"
+            role: slave
+            app: {{ template "redis.name" . }}
+      {{- end }}
```
Comment (juan131) [inline stable/redis/templates/networkpolicy.yaml:41]: You should close the previous `if` statement:
```diff
+     {{- end }}
      {{- if ge .Values.cluster.slaveCount 1.0 }}
      - podSelector:
          matchLabels:
            release: "{{ .Release.Name }}"
            role: slave
            app: {{ template "redis.name" . }}
-     {{- end }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the contribution @Yitaek 

Could you please take a look to my comments?

File: stable/redis/templates/networkpolicy.yaml (hunk 1 above)
Comment (Yitaek) [reply stable/redis/templates/networkpolicy.yaml:41]: @juan131 thanks for the catch. I updated the PR

**Comment by juan131** [comment]
Hi @Yitaek it seems you did not sign your last commit, could you please fix that?

**Comment by Yitaek** [comment]
@juan131 signed the last commit and re-pushed

**Review by juan131 — APPROVED** [review]
Thanks so much for addressing it!

**Pull request merged**